### ISSUE TITLE
Ripley cannot drill walls, or the drillchance is absurdly low.

### ISSUE THREAD

**Issue by Buggy123** [tgstation/tgstation#34426]
Step 1: Get ripley
Step 2: Put normal drill on it
Step 3: Mine wall
Step 4: Mine wall
Step 5: Mine wall
Step 6: Mine wall
Step 7: Mine wall
Step 8: Mine wall
Step 9: Mine wall
Step 10: Mine wall
Step 11: Mine wall
Step 12: Mine wall
Step 13: Mine wall
Step 14: Mine wall
Step 15: Mine wall
Step 16: Mine wall
Step 17: Mine wall
Step 18: Mine wall
Step 19: Mine airlock instead
Step 20: Airlock is instantly deleted 100% of the time

Happens consistently but a example round is ID 81883

**Comment by SpaceManiac** [comment]
I always thought this was intentional but it can drill blast doors so it does seem inconsistent.

**Comment by Hathkar** [comment]
Yeah, I've noticed this problem as well. Pretty sure mining walls doesn't work with a diamond drill either.

**Comment by vuonojenmustaturska** [comment]
For turfs, drills call drill_act(). /turf/closed/wall/r_wall, /turf/closed/mineral and /turf/open/floor/plating are the only types that implement drill_act(), for everything else nothing happens. Airlocks are not turfs and instead get ex_act(2) called on them.

**Comment by Buggy123** [comment]
So... It just doesn't do anything to 90% of the walls on the station?

**Comment by SpaceManiac** [comment]
Drilling rwalls but not walls seems like a strange inversion

**Comment by Magnnarot** [comment]
This problem is still ongoing, been a couple months since mechs can't drill any kind of walls. Seems to have extended to being unable to drill players in order to gib them.

**Comment by Xhuis** [comment]
I'm not related to this?
